docs/src/userguide/extension_types.rst

```diff
@@ -532,6 +532,100 @@ statically sized freelist of ``N`` instances for a given type.  Example::
 
 .. _making_extension_types_weak_referenceable:
 
+.. _existing-pointers-instantiation:
+
+Instantiation from existing C/C++ pointers
+===========================================
+
+It is quite common to want to instantiate an extension class from an existing
+(pointer to a) data structure, often as returned by external C/C++ functions.
+
+As extension classes can only accept Python objects as arguments in their
+contructors, this necessitates the use of factory functions. For example, ::
+
+    from libc.stdlib cimport malloc, free
+
+    # Example C struct
+    ctypedef struct my_c_struct:
+        int a
+        int b
+
+
+    cdef class WrapperClass:
+        """A wrapper class for a C/C++ data structure"""
+        cdef my_c_struct *_ptr
+        cdef bint ptr_owner
+
+        def __cinit__(self):
+            # On cinit, do not create new structure but set pointer to NULL
+            self._ptr = NULL
+            self.ptr_owner = True
```

Comment:
Given that the pointer is NULL here, I think it would make more sense if this was False by default.

Reply:
Updated.